File: server/java/src/main/java/com/hrms/business/abstracts/CandidateJobPostingFavoriteService.java
package com.hrms.business.abstracts;

import com.hrms.core.utilities.results.DataResult;
import com.hrms.core.utilities.results.Result;
import com.hrms.entities.concretes.CandidateJobPostingFavorite;

import java.util.List;

public interface CandidateJobPostingFavoriteService {
    Result save(CandidateJobPostingFavorite candidateJobPostingFavorite);
    Result delete(CandidateJobPostingFavorite candidateJobPostingFavorite);
    DataResult<List<CandidateJobPostingFavorite>> getAll();
    DataResult<List<CandidateJobPostingFavorite>> getAllByCandidateId(int candidateId);
    DataResult<CandidateJobPostingFavorite> getByCandidateIdAndJobPostingId(int candidateId,int jobPostingId);
}

File: client/react/hrms/src/services/candidateJobPostingFavoriteService.js
import axios from "axios";

export default class CandidateJobPostingFavoriteService {
  getAll() {
    return axios.get(
      "http://localhost:8080/api/candidatejobpostingfavorites/getall"
    );
  }

  getAllByCandidateId(candidateId) {
    return axios.get(
      "http://localhost:8080/api/candidatejobpostingfavorites/getallbycandidateid?candidateId=" +
        candidateId
    );
  }

  getByCandidateIdAndJobPostingId(candidateId, jobPostingId) {
    return axios.get(
      "http://localhost:8080/api/candidatejobpostingfavorites/getbycandidateidandjobpostingid?candidateId=" +
        candidateId +
        "&jobPostingId=" +
        jobPostingId
    );
  }

  save(candidateLink) {
    return axios.post(
      "http://localhost:8080/api/candidatejobpostingfavorites/save",
      candidateLink
    );
  }

  delete(candidateLink) {
    return axios.post(
      "http://localhost:8080/api/candidatejobpostingfavorites/delete",
      candidateLink
    );
  }
}
